### PR TITLE
Project refactor work

### PR DESCRIPTION
**Pull request by kerthcet**
What I did with this PR:
1. bump go version to v1.21 align with kubernetes
2. Rename foler internal to pkg as kueue does
3. Remove *.out and add a .gitignore file to this project.
4. change repo name leader-worker-set to lws
5. upgrade the controller-gen to v0.14.0 to fix test error and rerun `make manifest`.

fix https://github.com/kubernetes-sigs/lws/issues/13
part of https://github.com/kubernetes-sigs/lws/issues/8

### CODE REVIEW

**Comment by kerthcet** [comment]
cc @ahg-g can we merge?

**Comment by ahg-g** [comment]
> cc @ahg-g can we merge?

the test is failing, how did you verify that it passes? does it pass for you locally?

**Comment by kerthcet** [comment]
> > cc @ahg-g can we merge?
> 
> the test is failing, how did you verify that it passes? does it pass for you locally?

Yes, the test is always failing, I'm try to dig the reason, but I ran locally yesterday, the lws works as expected.

**Comment by kerthcet** [comment]
Let me fix the error first.

**Comment by kerthcet** [comment]
Test error is not related to the controller logic. Add a flag `-h` to `controller-gen` solved this, have no idea why this happens, tracking here https://github.com/kubernetes-sigs/controller-tools/issues/888. cc @ahg-g

File: Makefile [hunk 1]
```diff
@@ -49,11 +49,11 @@ help: ## Display this help.
 
 .PHONY: manifests
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) \
-	  rbac:roleName=manager-role output:rbac:artifacts:config=config/rbac\
-    crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/crd/bases\
-    webhook output:webhook:artifacts:config=config/webhook\
-    paths="./..."
+	$(CONTROLLER_GEN) -h \
```
Comment (ahg-g) [inline Makefile:52]: isn't this the help flag? which means it will actually not generate anything?

Comment (ahg-g) [reply Makefile:52]: I think the issue is that the project is using v0.13.0, we should upgrade to v0.14.0

Comment (ahg-g) [reply Makefile:52]: line 167 below change `CONTROLLER_TOOLS_VERSION ?= v0.13.0` to `CONTROLLER_TOOLS_VERSION ?= v0.14.0`

Comment (kerthcet) [reply Makefile]: Done. seems worked. Thanks

Comment (kerthcet) [reply Makefile]: Checked the changelog, https://github.com/kubernetes-sigs/controller-tools/releases, didn't see any related changes. how do you find that?

Comment (ahg-g) [reply Makefile]: I compared it with jobset

**Comment by ahg-g** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**